framework/test/AnyTest.cpp

```diff
@@ -149,5 +149,44 @@ int AnyTest(int /*argc*/, char* /*argv*/[])
   US_TEST_CONDITION(anyMap3.ToString() == "{map : {1 : 0.3, 3 : bye}, number : 5, vector : [9,8,7]}", "Any[std::map<std::string,Any>].ToString()")
   US_TEST_CONDITION(anyMap3.ToJSON() == "{\"map\" : {\"1\" : 0.3, \"3\" : \"bye\"}, \"number\" : 5, \"vector\" : [9,8,7]}", "Any[std::map<std::string,Any>].ToJSON()")
 
+  // Test BadAnyCastException exceptions
+  const Any uncastableConstAny(0.0);
+  Any uncastableAny(0.0);
+
+  US_TEST_FOR_EXCEPTION(cppmicroservices::BadAnyCastException, any_cast<std::string>(uncastableConstAny))
+  
+  try
```

Comment:
This capability could be added. I don't know if there is any real value in doing it with gtest unit tests coming soon.